src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -482,6 +482,11 @@ public ProcessStartInfo StartInfo
             {
                 if (_startInfo == null)
                 {
+                    if (Associated && Process.GetCurrentProcessId() != Id)
```

Comment:
You're right that environment variables is the only one thing that is have some useful meaning. So there no too much sense in returning this object. Some of the buggy information can be retrieved and fixed for example Arguments. What do you suggest also throw same exception for current process?

Please correct me if I wrong but does environment variables can be changed for running process. I found following on MSDN:
> Altering the environment variables of a child process during process creation is the only way one process can directly change the environment variables of another process. A process can never directly change the environment variables of another process that is not a child of that process.